src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -433,192 +395,163 @@ createBlockHistoryTable =
         \ endingtxid UNSIGNED BIGINT NOT NULL, \
         \ CONSTRAINT blockHashConstraint UNIQUE (blockheight));"
 
-createVersionHistoryTable  :: BlockHandler SQLiteEnv ()
-createVersionHistoryTable =
-    callDb "createVersionHistoryTable" $ \db ->
-    exec_ db
-        "CREATE TABLE IF NOT EXISTS VersionHistory (version UNSIGNED BIGINT NOT NULL,\
-        \ blockheight UNSIGNED BIGINT NOT NULL,\
-        \ txid UNSIGNED BIGINT NOT NULL,\
-        \ CONSTRAINT versionConstraint UNIQUE (version, blockheight));"
-
-createUserTablesTable  :: BlockHandler SQLiteEnv ()
-createUserTablesTable =
-    callDb "createUserTablesTable" $ \db ->
-    exec_ db
-        "CREATE TABLE IF NOT EXISTS UserTables (tablename TEXT NOT NULL\
-        \ , createBlockHeight UNSIGNED BIGINT NOT NULL\
-        \ , version UNSIGNED BIGINT NOT NULL\
-        \ , CONSTRAINT versionTableConstraint UNIQUE\
-        \ (version, createBlockHeight, tablename));"
-
-createUserTable :: TableName -> BlockVersion -> BlockHandler SQLiteEnv ()
-createUserTable name b = do
-    callDb "createUserTable" $ \db ->
-        exec_ db $
-            "CREATE TABLE IF NOT EXISTS "
-            <> Utf8 (toS (flip T.snoc ']' $ T.cons '[' $ (asString name)))
-            <> " (rowkey TEXT\
-               \, blockheight UNSIGNED BIGINT\
-               \, version UNSIGNED BIGINT\
-               \, txid UNSIGNED BIGINT\
-               \, rowdata BLOB \
-               \, UNIQUE(blockheight, rowkey, txid));"
-    versionedTablesInsert name b
-
-handleVersion :: BlockHeight -> ParentHash -> BlockHandler SQLiteEnv TxId
-handleVersion bRestore hsh = do
-    bCurrent <- getBCurrent
-    vCurrent <- getVCurrent
-    checkHistoryInvariant
-
-    case compare bRestore (bCurrent + 1) of
-        GT -> internalError "handleVersion: Block_Restore invariant violation!"
-        EQ -> newChildBlock bCurrent vCurrent
-        LT -> rewindBlock vCurrent
-
+createTableCreationTable :: BlockHandler SQLiteEnv ()
+createTableCreationTable =
+  callDb "createTableCreationTable" $ \db -> exec_ db
+    "CREATE TABLE VersionedTableCreation\
+    \(tablename TEXT NOT NULL\
+    \, createBlockheight UNSIGNED BIGINT NOT NULL\
+    \, CONSTRAINT creation_unique UNIQUE(tablename, createBlockheight));"
+
+createTableMutationTable :: BlockHandler SQLiteEnv ()
+createTableMutationTable =
+  callDb "createTableMutationTable" $ \db -> do
+  exec_ db "CREATE TABLE VersionedTableMutation\
+           \(tablename TEXT NOT NULL\
+           \, blockheight UNSIGNED BIGINT NOT NULL\
+           \, CONSTRAINT mutation_unique UNIQUE(tablename,blockheight));"
+  exec_ db "CREATE INDEX mutation_bh ON VersionedTableMutation(blockheight);"
+
+createUserTable :: TableName -> BlockHeight -> BlockHandler SQLiteEnv ()
+createUserTable name bh = do
+  let tablename :: Utf8
```

Comment:
usually prefer `where` to `let`-blocks in `do` bindings if the values are pure or only depend on the arguments

Reply:
done